registry/registry.go

```diff
@@ -133,9 +114,11 @@ func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 // DockerHeaders returns request modifiers that ensure requests have
 // the User-Agent header set to dockerUserAgent and that metaHeaders
 // are added.
-func DockerHeaders(metaHeaders http.Header) []transport.RequestModifier {
+func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.RequestModifier {
 	modifiers := []transport.RequestModifier{
-		transport.NewHeaderRequestModifier(http.Header{"User-Agent": []string{dockerUserAgent}}),
+		transport.NewHeaderRequestModifier(http.Header{
+			"User-Agent": []string{userAgent},
```

Comment:
I see many of the tests using it, I don't think it is something that needs to be enforced since golang can handle setting the header itself just fine.